cas-server-core-api-services/src/main/java/org/jasig/cas/services/RegisteredServiceAuthenticationPolicy.java

```diff
@@ -0,0 +1,44 @@
+package org.jasig.cas.services;
+
+import java.io.Serializable;
+import java.util.Set;
+
+/**
+ * This is {@link RegisteredServiceAuthenticationPolicy} that describes how a service
+ * should handle authentication requests.
+ *
+ * @author Misagh Moayyed
+ * @since 4.3.0
+ */
+public interface RegisteredServiceAuthenticationPolicy extends Serializable {
```

Comment:
You are right. There are authn policies associated with the authn manager and the CAS interface so I think it's best to really divorce this concept for that all and get it focused on MFA. I like `RegisteredServiceMultifactorPolicy` better which should convey the idea.